Rubeus/lib/Ask.cs

```diff
@@ -134,15 +134,17 @@ public static bool NoPreAuthTGT(string userName, string domain, string keyString
 
         public static byte[] TGT(string userName, string domain, string certFile, string certPass, Interop.KERB_ETYPE etype, string outfile, bool ptt, string domainController = "", LUID luid = new LUID(), bool describe = false, bool verifyCerts = false, string servicekey = "", bool getCredentials = false) {
             try {
-                X509Certificate2 cert;
+                X509Certificate2 cert = FindCertificate(certFile, certPass);
 
-                if (Helpers.IsBase64String(certFile))
+                // Check for Base64 encoded certificate second in case certFile was a hex-encoded fingerprint
+                if (cert == null && Helpers.IsBase64String(certFile))
                 {
                     cert = new X509Certificate2(Convert.FromBase64String(certFile), certPass);
                 }
-                else
-                {
-                    cert = FindCertificate(certFile, certPass);
+
+                if (cert == null) {
+                    Console.WriteLine("[!] Failed to find certificate for {0}", certFile);
+                    return null;
                 }

```

Comment:
```suggestion
```
Redundant `if (cert == null)` is duplicated in lines `145`-`148` and lines `150`-`153`.

Reply:
Good catch, not sure how that happened!